### ISSUE TITLE
bootstrapFunction props param type should be AppProps & CustomProps

### ISSUE THREAD

**Issue by thanhbinhtran93** [single-spa/single-spa-angular#281]
### Demonstration
When I register an application and pass customProps 

```
singleSpa.registerApplication({
  name: 'app1',
  activeWhen,
  app,
  customProps: (name, location) => {
    return { apiKey: "d83jD63UdZ6RS6f70D0" };
  }
});
```

app1 is an Angular 9 application, in which I want to use that `apiKey`

```
const lifecycles = singleSpaAngular({
  bootstrapFunction: singleSpaProps => {
    singleSpaPropsSubject.next(singleSpaProps);
    return platformBrowserDynamic([...getSingleSpaExtraProviders(), {provide: 'apiKey', useValue: singleSpaProps.apiKey}])
      .bootstrapModule(AppModule);
  },
  template: '<devices-root />',
  Router,
  NgZone,
});
```

### Expected Behavior
No Typescript error.

### Actual Behavior
I get the error `Property 'apiKey' does not exist on type 'AppProps'.`

The `singleSpaProps` param in the `bootstrapFunction` includes both the AppProps and CustomProps; therefore, its type should also be `AppProps & CustomProps`

dependencies in package.json
```

"dependencies": {
    "@angular/animations": "~9.1.11",
    "@angular/common": "~9.1.11",
    "@angular/compiler": "~9.1.11",
    "@angular/core": "~9.1.11",
    "@angular/forms": "~9.1.11",
    "@angular/platform-browser": "~9.1.11",
    "@angular/platform-browser-dynamic": "~9.1.11",
    "@angular/router": "~9.1.11",
    "@ngx-translate/core": "^12.1.2",
    "rxjs": "~6.5.4",
    "single-spa": ">=4.0.0",
    "single-spa-angular": "^4.3.1",
    "tslib": "^1.10.0",
    "zone.js": "~0.10.2"
  },
```

**Comment by joeldenning** [comment]
Hi @thanhbinhtran93, it looks like we need to add generics to the `singleSpaAngular` function:

```ts
const lifecycles = singleSpaAngular<MyProps>({
  bootstrapFunction: singleSpaProps => {
    singleSpaPropsSubject.next(singleSpaProps);
    return platformBrowserDynamic([...getSingleSpaExtraProviders(), {provide: 'apiKey', useValue: singleSpaProps.apiKey}])
      .bootstrapModule(AppModule);
  },
  template: '<devices-root />',
  Router,
  NgZone,
});

type MyProps = {
  apiKey: string;
}
```

Here's the line of code that should change:

https://github.com/single-spa/single-spa-angular/blob/f0facdc61a0ce9492f9eb7fa7a5233414f8b08aa/src/src/single-spa-angular.ts#L23

Would you be interested in submitting a pull request to fix this?

**Comment by thanhbinhtran93** [comment]
@joeldenning Sure! I can do the pull request this week.

Since the singleSpa custom props is defined here:
https://github.com/single-spa/single-spa-angular/blob/f0facdc61a0ce9492f9eb7fa7a5233414f8b08aa/schematics/ng-add/_files/src/single-spa/single-spa-props.ts.template#L8

I'm thinking about also making the change in schematics  e.g.

```
export type SingleSpaCustomProps = {};

export type SingleSpaProps = AppProps & SingleSpaCustomProps;
```

Then in main.single-spa.ts.template:

```
const lifecycles = singleSpaAngular<SingleSpaCustomProps>({
  ...
});
```

**Comment by joeldenning** [comment]
The custom props are already generic in https://github.com/single-spa/single-spa/blob/a8d6c8441d2e948d9e654229116479e7bc26e825/typings/single-spa.d.ts#L52

I think the signature for the singleSpaAngular function should be the following:

```ts
 export function singleSpaAngular<T>(userOptions: SingleSpaAngularOptions): LifeCycles<T> { 
```

**Comment by thanhbinhtran93** [comment]
That's not enough since the bootstrapFunction also needs to know about the custom prop. The signature for bootstrapFunction should be: 
```
bootstrapFunction(props: AppProps & T): Promise<NgModuleRef<any>>;
```

The bootstrapFunction is defined inside BaseSingleSpaAngularOptions:
https://github.com/single-spa/single-spa-angular/blob/f0facdc61a0ce9492f9eb7fa7a5233414f8b08aa/src/internals/src/types.ts#L6-L10

So the BaseSingleSpaAngularOptions would become
```
export interface BaseSingleSpaAngularOptions<T> {
  ...
  bootstrapFunction(props: AppProps & T): Promise<NgModuleRef<any>>;
}
```

However, the BaseSingleSpaAngularOptions is used in multiple places, I don't know yet how to make the change efficiently.

**Comment by joeldenning** [comment]
Resolved in #283
